gpMgmt/bin/gppylib/logfilter.py

```diff
@@ -273,6 +273,17 @@ def next(self):
         #we need to make a minor format change to the log level field so that
         # our single regex will match both.
         item[16] = item[16] + ": "
+
+        # There is a text containing \n or | symbols. We need to add quotation 
+        # marks, for parsers to process csv text correctly. We also need to 
+        # double quote in places where they already are.
+        for i in range(len(item)):
+            if '\n' in item[i] or '|' in item[i]:
+                if '\"' in item[i]:
+                    item[i] = item[i].replace('\"', '\"\"')
```

Comment:
If item contains double-quote we have to quote overall string anyway. You could check the result of exporting csv line without enclosing quotes - postgres removes internal quotes